dabench/dacycler/_etkf.py

```diff
@@ -61,96 +62,34 @@ def __init__(self,
-        H = H.at[jnp.arange(H.shape[0]), 
-                obs_loc_indices.flatten()
-                 ].set(1)
-        return H
-
-    def _calc_default_R(self, obs_values, obs_error_sd):
-        return jnp.identity(obs_values.flatten().shape[0])*(obs_error_sd**2)
-
-    def _calc_default_B(self):
-        return jnp.identity(self.system_dim)
-
-    def _cycle_obsop(self, Xbt, obs_values, obs_loc_indices, obs_error_sd,
-                     obs_time_mask, obs_loc_mask,
-                     H=None, h=None, R=None, B=None):
-        if H is None and h is None:
-            if self.H is None:
-                if self.h is None:
-                    H = self._calc_default_H(obs_values, obs_loc_indices)
-                else:
-                    h = self.h
-            else:
-                H = self.H
-        if R is None:
-            if self.R is None:
-                R = self._calc_default_R(obs_values, obs_error_sd)
-            else:
-                R = self.R
-        if B is None:
-            if self.B is None:
-                B = self._calc_default_B()
-            else:
-                B = self.B
-
-        nr, nc = Xbt.shape
-        assert nr == self.ensemble_dim, (
-                'cycle:: model_forecast must have dimension {}x{}').format(
-                    self.ensemble_dim, self.system_dim)
-
-        # Apply obs masks to H
-        H = jnp.where(obs_time_mask, H.T, 0).T
-        H = jnp.where(obs_loc_mask.flatten(), H.T, 0).T
-
-        # Analysis cycles over all obs in data_obs
-        Xa = self._compute_analysis(Xb=Xbt.T,
-                                    y=obs_values,
-                                    H=H,
-                                    h=h,
-                                    R=R,
-                                    rho=self.multiplicative_inflation)
-
-        return Xa.T, 0
-
     def _step_forecast(self, xa, n_steps):
-        data_forecast = []
+        """Ensemble method needs a slightly different _step_forecast method"""
+        ensemble_forecasts = []
+        ensemble_inputs = []
         for i in range(self.ensemble_dim):
-            new_vals = self.model_obj.forecast(
-                    vector.StateVector(values=xa.values[i], store_as_jax=True),
+            cur_inputs, cur_forecast = self.model_obj.forecast(
+                    xa.isel(ensemble=i),
                     n_steps=n_steps
-                    ).values
-            data_forecast.append(new_vals)
+                    )
+            ensemble_inputs.append(cur_inputs)
+            ensemble_forecasts.append(cur_forecast)
 
-        out_vals = jnp.moveaxis(jnp.stack(data_forecast), [0,1,2],[1,0,2])
-        return vector.StateVector(values=out_vals,
-                                  store_as_jax=True)
+        return (xr.concat(ensemble_inputs, dim='ensemble'),
+                xr.concat(ensemble_forecasts, dim='ensemble'))
 
-    def _apply_obsop(self, Xb, H, h):
+    def _apply_obsop(self, xb, H, h):
```

Comment:
@kysolvik why are you changing all of the matrices to lower case vector notation here?